packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3071,6 +3096,10 @@ class NavigatorState extends State<Navigator> with TickerProviderStateMixin {
       entry = previous;
       previous = index > 0 ? _history[index - 1] : null;
     }
+
+    // Inform navigator observers about route changes.
+    _informObservers();
```

Comment:
nit: maybe call this flushObserverNotifications? (similarly to the _flushRouteAnnouncement below).